presto-main/src/main/java/com/facebook/presto/sql/planner/PlanOptimizers.java

```diff
@@ -258,6 +260,19 @@ public PlanOptimizers(
                 new MetadataQueryOptimizer(metadata),
                 new EliminateCrossJoins(), // This can pull up Filter and Project nodes from between Joins, so we need to push them down again
                 new PredicatePushDown(metadata, sqlParser),
+                new PruneUnreferencedOutputs(),
+                new IterativeOptimizer(
```

Comment:
add comments why this rule is needed before join reordering (also mention `PredicatePushDown` as it's probably also required in order to have optimal join enumeration)